File: AutomaticTimetableGeneratorSystem-main/timetable-backend/src/main/java/com/timetable/service/SubjectWorkloadService.java
package com.timetable.service;

import java.util.List;

import com.timetable.dto.subjectworkload.SubjectWorkloadRequest;
import com.timetable.dto.subjectworkload.SubjectWorkloadResponse;

public interface SubjectWorkloadService {

    /**
     * Save Subject Workload
     */
    SubjectWorkloadResponse saveSubjectWorkload(
            SubjectWorkloadRequest request);

    /**
     * Update Subject Workload
     */
    SubjectWorkloadResponse updateSubjectWorkload(
            Long workloadId,
            SubjectWorkloadRequest request);

    /**
     * Delete Subject Workload
     */
    void deleteSubjectWorkload(Long workloadId);

    /**
     * Get Subject Workload By ID
     */
    SubjectWorkloadResponse getSubjectWorkloadById(
            Long workloadId);

    /**
     * Get All Subject Workloads
     */
    List<SubjectWorkloadResponse> getAllSubjectWorkloads();

    /**
     * Get Subject Workload By Allocation
     */
    SubjectWorkloadResponse getSubjectWorkloadByAllocation(
            Long allocationId);
}

File: AutomaticTimetableGeneratorSystem-main/timetable-frontend/src/services/subjectWorkloadService.js
import api from "./api";

const BASE_URL = "/subject-workloads";

const subjectWorkloadService = {

    // Get All Subject Workloads
    getAllSubjectWorkloads: async () => {

        const response = await api.get(BASE_URL);

        return response.data;

    },

    // Get Subject Workload By Id
    getSubjectWorkloadById: async (id) => {

        const response = await api.get(
            `${BASE_URL}/${id}`
        );

        return response.data;

    },

    // Get Subject Workload By Allocation
    getSubjectWorkloadByAllocation: async (allocationId) => {

        const response = await api.get(
            `${BASE_URL}/allocation/${allocationId}`
        );

        return response.data;

    },

    // Create
    saveSubjectWorkload: async (data) => {

        const response = await api.post(
            BASE_URL,
            data
        );

        return response.data;

    },

    // Update
    updateSubjectWorkload: async (id, data) => {

        const response = await api.put(
            `${BASE_URL}/${id}`,
            data
        );

        return response.data;

    },

    // Delete
    deleteSubjectWorkload: async (id) => {

        const response = await api.delete(
            `${BASE_URL}/${id}`
        );

        return response.data;

    }

};

export default subjectWorkloadService;